ptvsd/wrapper.py

```diff
@@ -990,8 +990,11 @@ def on_threads(self, request, args):
 
         try:
             xml = self.parse_xml_response(resp_args)
-        except SAXParseException:
-            self.send_response(request, success=False)
+        except SAXParseException as ex:
+            self.send_response(
+                request,
+                success=False,
+                message=ex.getMessage())
```

Comment:
`getMessage()`!? Never seen that before. Hmm, I always thought the best way was to use `sys.exec_info()` (it's a tuple).
Best to wait for someone (@int19h) else to review this.
I'd create a helper method that sends the error `this.send_error_response(request)` (this way we have the opportunity to log the error and change what is sent back, error type, message, stack trace...)

Reply:
This is specific to the exception that we're catching here (for the SAX parser). But I don't think we actually want to propagate that message, if it surfaces anywhere in UI, which IIRC it does. I mean, this would be something like "malformed XML", showing up in Locals - even though the user has no idea what it has to do with XML.

We could just pass the text that we failed to parse - that would say something like "frame ... not found". Still not perfect, but probably better, if we want to say something there.

PyDev seems to be using a generic error message when it can't retrieve frame variables.

Reply:
> This is specific to the exception that we're catching here 

Aah yes,

Reply:
I think `this.send_error_response(request)` is a good idea. I will add that. If we have a generalized error response then `sys.exec_info()` would be a better idea. For specific cases, , we may be sending back information which may not be helpful. for `SAXExceptions` they have `getMessage()` which return human readable errors: https://docs.python.org/2/library/xml.sax.html

Reply:
As discussed, I need to remove the `getMessage` I will do that before merging.